wrapper/src/test/java/integration/refactored/container/tests/AutoscalingTests.java

```diff
@@ -109,6 +109,9 @@ protected HikariPoolConfigurator getHikariConfig(int maxPoolSize) {
   @TestTemplate
   public void test_pooledConnectionAutoScaling_setReadOnlyOnOldConnection()
       throws SQLException, InterruptedException {
+    // Timeout controlling whether a cached connection should be checked for validity before being returned.
+    // Replicate worst-case scenario: hikari does not check for validity before returning cached connections.
+    System.setProperty("com.zaxxer.hikari.aliveBypassWindowMs", String.valueOf(TimeUnit.HOURS.toMillis(1)));
```

Comment:
Why is this timeout 1 hour compared to the 10 minutes in the r/w splitting tests?

Reply:
We want to set this setting to some value greater than the amount of time the tests take so that we can replicate that worst-case scenario. The r/w splitting tests take <5 minutes each usually, but the autoscaling tests take ~20 minutes each